### DISCUSSION TITLE
GENERATED PDF FILE IS EMPTY WKHTMLTOPDF

### DISCUSSION THREAD

**Discussion by ahmadfauzanarif** [serenity-is/Serenity#6343] [Help] [answered]
![Screenshot_13](https://user-images.githubusercontent.com/71299117/160998712-0e46e010-7e23-480b-8e8e-264c9641f0bb.png)

Hi, I want to ask. I got an error after rendering PDF using WKHTMLTOPDF, previously I was able to render using WKHTMLTOPDF and it worked fine, but after I changed to the latest version of Serenity Community, I got an error, where the error is GENERATED PDF FILE IS EMPTY. Thank you, hopefully someone will experience this and help me

**Comment by asilmertkan** [comment, marked as answer]
I had this problem too. Finally I found a way. Check your appsettings.json for EnvironmentSettings and write your Url like this 
for local 
"EnvironmentSettings": {
    "SiteExternalUrl": "https://localhost:yourport/"
  },
for publish
 "EnvironmentSettings": {
    "SiteExternalUrl": "yourexternalurl/"
  },
 and after add this pasage to your webconfig
for local
```
<appSettings>
		<add key="EnvironmentSettings" value="{ SiteExternalUrl: 'https://localhost:yourport/' }" />
	</appSettings>
```
for publish
```
<appSettings>
		<add key="EnvironmentSettings" value="{ SiteExternalUrl: 'yourexternalurl/' }" />
	</appSettings>
```

If you have problem still, you must sure wkhtmltopdf.exe (true version) in your solution main folder. I hope this will be helpful.

**Reply by ahmadfauzanarif** [reply]
![image](https://user-images.githubusercontent.com/71299117/188052382-40af4751-7d0b-402e-978c-865151560f68.png)


Thank you very much it works!

**Reply by VictorTomaili** [reply]
@ahmadfauzanarif please remove or blur the phone number email and address information from your images.

**Reply by ahmadfauzanarif** [reply]
@VictorTomaili Done

**Comment by rhemasystems** [comment]
Mine works in the VPS when running it internally in the server but not outside the server
